waku/factory/node_factory.nim

```diff
@@ -352,11 +352,11 @@ proc setupProtocols(
     except CatchableError:
       return err("failed to mount waku filter protocol: " & getCurrentExceptionMsg())
 
+  await node.mountFilterClient()
```

Comment:
Same with filter.

Reply:
Yes indeed, good point!, and I agree that it may sound weird but it is handier from user PoV to always have the "client" components mounted so that is not needed an additional argument such as `filter-client = true`, or `lightpush-client = true`, or `store-client

Reply:
Well, imho, something more robust configuration option maybe needed. Like --light-protocols (edge-protocols?)
then install filter/lightpush/store-clients. Also we can leave the option to specify the service node(s) for the client protocols - but non mandatory. I think it can be a good balance between ease of use and control over resources.

I just noticed that with this change REST API will not function as expected. filter and lightpush REST endpoints are initialized upon --filternode and --lightpushnode having value.
Then this needs to be added here to keep consistency.
WDYT @Ivansete-status ?

Reply:
@NagyZoltanPeter - yes it makes 100% sense to me to add that `--light-protocols` option. That's something we will need shortly. I think we can consider that in a separate PR.

Reply:
Ok fine for me. I will approve this PR now!